test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/ViewComponentMethodSelectorTest.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    public class ViewComponentMethodSelectorTest
+    {
+        [Theory]
+        [InlineData(typeof(ViewComponentWithSyncInvoke), new object[0])]
+        [InlineData(typeof(ViewComponentWithSyncInvoke), new object[] { "" })]
+        [InlineData(typeof(ViewComponentWithSyncInvoke), new object[] { 42 })]
+        [InlineData(typeof(ViewComponentWithAsyncInvoke), new object[] { 42, false })]
```

Comment:
Isn't this the only data that leads to failure due to type mismatches? Put another way, looks like all `[InlineData]` using `ViewComponentWithSyncInvoke` are the same test. Either that or the data for this test and `FindSyncMethod_ReturnsNull_IfMatchCannotBeFound()` were switched at birth.